docs/dev-env.md

```diff
@@ -4,6 +4,10 @@
 
 Clone this repository locally, with `git clone <repo-url>`, then `cd <repo-dir>`,
 
+## MacOS
+
+If you're installing on macOS, refer to the [macOS Guide](./macos.md) instead of this file.
+
```

Comment:
Could you change this to just say "If you encounter problems with Nix on MacOS, refer to the [MacOS Guide](./macos.md)"

Nix should work on Mac. It just breaks a little more on Mac than on other operating systems in our experience.